### PR TITLE
new prototype guidance pages with screenshots

### PR DESCRIPTION
**Pull request by karypun**
- updated get started page
- starting your prototype
- using components
- deploying (placeholder)

### CODE REVIEW

**Comment by daniel-ac-martin** [comment]
@karypun: Looks like it's complaining about `class` being used instead of `className`:
> Property 'class' does not exist on type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.

**Comment by karypun** [comment]
> @karypun: Looks like it's complaining about `class` being used instead of `className`:
> 
> > Property 'class' does not exist on type 'DetailedHTMLProps<HTMLAttributes, HTMLDivElement>'.

Was it from this? https://not-govuk.netlify.app/components?name=Inset%20text&show-InsetText-example=html I copied the HTML code snippet from the Inset text component.

I've updated the code from 'class' to 'className' and raised an issue to update HTML in NotGovUK

**Comment by daniel-ac-martin** [comment]
Preview: https://60071914ba3b853b8b923cba--hods-poc.netlify.app/get-started

**Comment by daniel-ac-martin** [comment]
Looks pretty good although I don't think we should stretch the images.

File: apps/docs/src/common/pages/get-started/deploy-prototype.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { FC, Fragment, createElement as h } from 'react';
+import { Helmet } from 'react-helmet-async';
+import { PageProps } from '@not-govuk/app-composer';
+import { A } from '@not-govuk/components';
+
+const Page: FC<PageProps> = ({ location }) => (
+  <div className="govuk-grid-row">
+    <Helmet>
+      <title>Guidance on Home Office Design System</title>
+      <meta name="og:article:section" content="Resources" />
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/deploy-prototype.tsx]: Replace `Resources` with `Get started`.

File: apps/docs/src/common/pages/get-started/project.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { PageProps } from '@not-govuk/app-composer';
+import { A } from '@not-govuk/components';
+
+
+
+const Page: FC<PageProps> = ({ location }) => (
+  <div className="govuk-grid-row">
+    <Helmet>
+      <title>Prototyping - Home Office Design System</title>
+      <meta name="og:article:section" content="Resources" />
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/project.tsx]: Replace `Resources` with `Get started`.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import { A } from '@not-govuk/components';
+const applyPrototype1 = require('../../../../assets/images/prototype-screen01.png').default;
+const applyPrototype2 = require('../../../../assets/images/prototype-screen02.png').default;
+
+
+const Page: FC<PageProps> = ({ location }) => (
+  <div className="govuk-grid-row">
+    <Helmet>
+      <title>Guidance on Home Office Design System</title>
+      <meta name="og:article:section" content="Resources" />
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: Replace `Resources` with `Get started`.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      application.
+      </div>
+
+      <h2>Before you start</h2>
+        <p>You'll need:</p>
+        <p><A href="https://nodejs.org/en/">Node.js</A> - version 12.x.x or higher</p>
+
+        <p>For the purpose of setting up the prototype, we are assuming you'll be using a Mac which comes with a terminal application to install, start and stop the prototype. Using a terminal is sometimes called ‘using the command line’. </p>
+
+      <h2>Installing the prototype</h2>
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: "Generating a new prototype"

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        <h3>1. Run the following commands locally</h3>
+        <p>Using terminal, run the following commands below (the parts in ALL CAPS should be replaced with something specific to you):</p>
+
+        <pre><code>
+        mkdir YOUR-PROTOTYPE
+        cd YOUR-PROTOTYPE

+        git init

+        npm init @hods prototype
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: Looks like these lines each need a `<br />` at the end of them.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+          <img  src={applyPrototype2} className="confirm-desktop" style={{width: "100%", maxHeight: "100%"}} alt="Terminal screen with a highlights around npm run dev command and no issues found"/>
+          </div>
+        </div>
+        <p>You can see the prototype by opening <A href="http://localhost:8080/">http://localhost:8080/</A> in your web browser.</p>
+        <h3>4. Optional: Set up your remote repository</h3>
+          <p>If you want to deploy your prototype so it can be tested with users or allow others to collaborate on it, you can set up a blank code repository in GitHub</p>
+        <h3>5. Optional: Commit and push your new prototype</h3>
+          <p>Run the following commands to push up the first version of your new prototype to your remote code repository:</p>
+        <pre><code>
+        git remote add origin git@github.com:YOUR_USER/YOUR-PROTOTYPE.git
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: Needs `<br />` at the end of the line.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...

+        npm init @hods prototype
+        </code></pre>
+
+        <h3>2. Answer the interactive prompts</h3>
+        <p>The generator / initialiser will ask you some questions but the defaults should be fine. You can accept the default by simply pressing enter in the terminal.</p>
+
+        <div className="govuk-grid-row">
+          <div className="govuk-grid-column-full">
+        <img  src={applyPrototype1} className="confirm-desktop" style={{width: "100%", maxHeight: "100%"}} alt="Terminal screen with a highlight around running commands to set up prototype"/>
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: I think we should remove the `width: 100%`. - The image doesn't look good when stretched.

**Comment by daniel-ac-martin** [comment]
I also wonder if modifying the screenshots is helpful. - It makes them look less like screenshots.

**Comment by karypun** [comment]
> I also wonder if modifying the screenshots is helpful. - It makes them look less like screenshots.

I modified it to remove my personal details from the guidance.  Addressed the other changes. Let me know what you think.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        <pre><code>
+        git remote add origin git@github.com:YOUR_USER/YOUR-PROTOTYPE.git <br />
+        git push -u origin master
+        </code></pre>
+        </div>
+  </div>
+);
+
+export default Page;
+export const title = 'Get started';
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: `'Starting a new prototype'`

File: apps/docs/src/common/pages/get-started/use-prototype.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            <img  src={applyPrototype5} className="confirm-desktop" style={{maxHeight: "100%"}} alt="Local web browser displaying properly coded status banner"/>
+          </div>
+        </div>
+
+    </div>
+  </div>
+);
+
+export default Page;
+export const title = 'Get started';
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/use-prototype.tsx]: `'Using the components'`

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        <p><A href="https://nodejs.org/en/">Node.js</A> - version 12.x.x or higher</p>
+
+        <p>For the purpose of generating a new prototype, we are assuming you'll be using a Mac which comes with a terminal application to install, start and stop the prototype. Using a terminal is sometimes called ‘using the command line’. </p>
+
+      <h2>Installing the prototype</h2>
+        <h3>1. Run the following commands locally</h3>
+        <p>Using terminal, run the following commands below (the parts in ALL CAPS should be replaced with something specific to you):</p>
+
+        <pre><code>
+        mkdir YOUR-PROTOTYPE <br />
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: The space before the `<br />` should be removed but it's not that big of a deal.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx [hunk 11]
```diff
@@ -10,7 +10,7 @@ const Page: FC<PageProps> = ({ location }) => (
   <div className="govuk-grid-row">
     <Helmet>
       <title>Guidance on Home Office Design System</title>
-      <meta name="og:article:section" content="Get started" />
+      <meta name="og:article:section" content="Starting a new prototype" />
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/start-prototype.tsx]: This was actually correct to begin with. - It's the section that the page sits in.

File: apps/docs/src/common/pages/get-started/use-prototype.tsx [hunk 12]
```diff
@@ -10,7 +10,7 @@ const Page: FC<PageProps> = ({ location }) => (
   <div className="govuk-grid-row">
     <Helmet>
       <title>Guidance on Home Office Design System</title>
-      <meta name="og:article:section" content="Get started" />
+      <meta name="og:article:section" content="Using the components" />
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/get-started/use-prototype.tsx]: This was actually correct to begin with. - It's the section that the page sits in.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx (hunk 11 above)
Comment (daniel-ac-martin) [reply apps/docs/src/common/pages/get-started/start-prototype.tsx]: The bit that needed to change was the `export const` line at the bottom.

File: apps/docs/src/common/pages/get-started/use-prototype.tsx (hunk 12 above)
Comment (daniel-ac-martin) [reply apps/docs/src/common/pages/get-started/use-prototype.tsx]: The bit that needed to change was the `export const` line at the bottom.

File: apps/docs/src/common/pages/get-started/start-prototype.tsx (hunk 11 above)
Comment (karypun) [reply apps/docs/src/common/pages/get-started/start-prototype.tsx]: sorry, I've just amended this

File: apps/docs/src/common/pages/get-started/use-prototype.tsx (hunk 12 above)
Comment (karypun) [reply apps/docs/src/common/pages/get-started/use-prototype.tsx]: Amended this now

File: apps/docs/src/common/pages/get-started/start-prototype.tsx (hunk 8 above)
Comment (karypun) [reply apps/docs/src/common/pages/get-started/start-prototype.tsx]: This has now been updated.

File: apps/docs/src/common/pages/get-started/use-prototype.tsx (hunk 9 above)
Comment (karypun) [reply apps/docs/src/common/pages/get-started/use-prototype.tsx]: Updated.

**Review by daniel-ac-martin — APPROVED** [review]

**Comment by daniel-ac-martin** [comment]
New preview: https://600d9ec611748ada4075307a--hods-poc.netlify.app/

**Pull request merged**